lib/usePantry.ts

```diff
@@ -334,6 +343,42 @@ async function handleComplexVersions(versions: PlainObject): Promise<SemVer[]> {
   return rv
 }
 
+async function handleURLVersions(versions: PlainObject): Promise<SemVer[]> {
+  const rv: SemVer[] = []
+  const url = validate_str(versions.url)
+  const matcher = validate_str(versions.match)
+
+  const body = await fetch(url).then(x => x.text())
+  const matches = body.matchAll(new RegExp(matcher.slice(1, -1), 'g'))
+
+  const strip = versions.strip
+  for (const match of matches) {
+    let m = ((x: string) => {
+      if (!strip) return x
+      if (isString(strip)) return x.replace(new RegExp(strip.slice(1, -1)), "")
+      if (isArray(strip)) {
+        for (const rx of strip) {
+          x = x.replace(new RegExp(rx.slice(1, -1)), "")
+        }
+        return x
+      }
+      throw new Error()
+    })(match[0])
+
+    // We'll handle dates > calver automatically. For now.
```

Comment:
possibly a nice feature for tags too…?

Reply:
I think allowing `semver.parse` to try the same string with `.replaceAll("-", ".")` in `tolerant` mode will probably fix more problems than it causes. Some repos have tags like `project_3_1_22`.